src/components/Event/components/EditableTimeField.tsx

```diff
@@ -0,0 +1,252 @@
+      pickerContext.setOpen(false);
+      // Clear focus after dropdown selection
+      setIsFocused(false);
+      inputRef.current?.blur();
+    }
+
+    prevFormattedValueRef.current = newFormattedValue;
+  }, [getFormattedValue, isFocused, pickerContext]);
+
+  const wasOpenRef = useRef(pickerContext.open);
+
+  // Handle dropdown open/close
+  useEffect(() => {
+    const wasOpen = wasOpenRef.current;
+    const isOpen = pickerContext.open;
+
+    if (isOpen && !wasOpen && inputRef.current) {
+      // Dropdown just opened - refocus input to allow typing
+      const timer = setTimeout(() => {
+        inputRef.current?.focus();
+      }, 10);
+      wasOpenRef.current = isOpen;
+      return () => clearTimeout(timer);
+    }
+
+    if (!isOpen && wasOpen) {
+      // Dropdown just closed - sync input value and clear focus state
+      setInputValue(getFormattedValue());
+      setIsFocused(false);
+      // Blur input to clear DOM focus (removes Mui-focused class)
+      inputRef.current?.blur();
+    }
+
+    wasOpenRef.current = isOpen;
+  }, [pickerContext.open, getFormattedValue]);
+
+  const { hasValidationError } = useValidation({
+    validator,
+    value: pickerContext.value,
+    timezone: pickerContext.timezone,
+    props: internalProps,
+  });
+
+  const parseAndUpdateTime = useCallback(
+    (value: string) => {
+      const trimmed = value.trim();
+      if (!trimmed) {
+        setInputValue(getFormattedValue());
+        return;
+      }
+
+      const newValue = parseTimeInput(
+        trimmed,
+        pickerContext.value as Dayjs | null
+      );
+
+      if (newValue) {
+        // Update picker internal state
+        pickerContext.setValue(newValue, {
+          validationError: null,
+        });
+
+        // Call onChange directly to ensure parent component is notified
+        if (onChange) {
+          onChange(newValue);
+        }
+
+        setInputValue(newValue.format(TIME_DISPLAY_FORMAT));
+      } else {
+        // Invalid input - reset to current value
+        setInputValue(getFormattedValue());
+      }
+    },
+    [pickerContext, getFormattedValue, onChange]
+  );
+
+  const handleClick = (e: React.MouseEvent) => {
+    // Stop propagation to prevent parent from toggling the picker
+    e.stopPropagation();
+
+    // Always keep it open when clicked, or open it if closed
+    if (!pickerContext.open) {
+      pickerContext.setOpen(true);
+    }
+  };
+
+  const handleFocus = () => {
+    setIsFocused(true);
+  };
+
+  const handleBlur = (e: React.FocusEvent<HTMLInputElement>) => {
+    // Check if blur is caused by clicking on dropdown
+    const relatedTarget = e.relatedTarget as HTMLElement | null;
+    const isClickingDropdown =
+      relatedTarget?.closest(".MuiPickerPopper-root") ||
+      relatedTarget?.closest(".MuiMultiSectionDigitalClock-root") ||
+      relatedTarget?.closest('[role="listbox"]') ||
+      relatedTarget?.closest('[role="option"]');
```

Comment:
big big red flag. 

I didnt take the time to read all the API, but I'm pretty sure we can do without all this hacky solution